rust/blockstore/src/arrow/sparse_index.rs

```diff
@@ -535,29 +567,35 @@ impl SparseIndex {
                 }
             };
             forward.insert(delimiter.clone(), block_id);
-            reverse.insert(block_id, delimiter);
             i += 1;
         }
         Ok(Self {
-            data: Arc::new(Mutex::new(SparseIndexData { forward, reverse })),
+            data: Arc::new(SparseIndexReaderData { forward }),
             id,
         })
     }
 }
 
-impl Debug for SparseIndex {
+impl Debug for SparseIndexReader {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let data = self.data.lock();
-        f.debug_struct("SparseIndex")
+        f.debug_struct("SparseIndexReader")
             .field("id", &self.id)
-            .field("data", &data)
             .finish()
     }
 }
 
-impl chroma_cache::Weighted for SparseIndex {
-    fn weight(&self) -> usize {
-        1
+pub(super) fn get_target_block_id(
```

Comment:
can we add a comment here that it is the caller's responsibility to ensure that the `forward` btree is synchronized properly if needed?

Reply:
this should be internal to the module, its just a helper. i'll address.

Reply:
https://github.com/chroma-core/chroma/pull/2918/commits/f77b8782d747e97b8a1bbc2c39287552f47a7510